From 2c66275b801a0846a97ffc5121154255d20c18d1 Mon Sep 17 00:00:00 2001
From: wlq <wlq@rock-chips.com>
Date: Sat, 12 Oct 2013 16:21:45 +0800
Subject: [PATCH] androidComputer: changed for display 	[modified file] 
 	arch/arm/mach-rk3188/board-rk3188-ac.c       	
 drivers/misc/ac_usb_switch.c 	       	drivers/video/rockchip/screen/Kconfig 
       		drivers/video/rockchip/screen/Makefile        	
 include/linux/input.h        		include/linux/scaler-core.h

---
 arch/arm/mach-rk3188/board-rk3188-ac.c | 2 +-
 drivers/misc/ac_usb_switch.c           | 4 ++--
 drivers/video/rockchip/screen/Kconfig  | 3 +++
 drivers/video/rockchip/screen/Makefile | 1 +
 include/linux/input.h                  | 1 -
 include/linux/scaler-core.h            | 2 +-
 6 files changed, 8 insertions(+), 5 deletions(-)

diff --git a/arch/arm/mach-rk3188/board-rk3188-ac.c b/arch/arm/mach-rk3188/board-rk3188-ac.c
index 5feb14a990ca..6805fbe04f02 100755
--- a/arch/arm/mach-rk3188/board-rk3188-ac.c
+++ b/arch/arm/mach-rk3188/board-rk3188-ac.c
@@ -96,7 +96,7 @@ static struct rk29_keys_button key_button[] = {
 	*/
 	{
                 .desc   = "mode_switch",
-                .code   = KEY_MODE_SWITCH,
+                .code   = 248,
                 //.desc   = "vol+",
                 //.code   = KEY_VOLUMEUP,
                 
diff --git a/drivers/misc/ac_usb_switch.c b/drivers/misc/ac_usb_switch.c
index a72bcd4663a0..a083c28a5544 100755
--- a/drivers/misc/ac_usb_switch.c
+++ b/drivers/misc/ac_usb_switch.c
@@ -20,8 +20,8 @@
 #else
 #define DBG(x...)
 #endif
-#define PC_DISPLAY_MODE 1 
-#define ANDROID_DISPLAY_MODE   2
+#define PC_DISPLAY_MODE 2
+#define ANDROID_DISPLAY_MODE   1
 
 #define USB_SWITCH_IOCTL_BASE 'u'
 #define USB_SWITCH_IOCTL_SET_USB_SWITCH_MODE		_IOW(USB_SWITCH_IOCTL_BASE, 0x01, int)
diff --git a/drivers/video/rockchip/screen/Kconfig b/drivers/video/rockchip/screen/Kconfig
index ac08545bc105..9f5ed6df7bcc 100755
--- a/drivers/video/rockchip/screen/Kconfig
+++ b/drivers/video/rockchip/screen/Kconfig
@@ -162,6 +162,9 @@ config LCD_HSD07PFW1
 config LCD_I30_800X480
         bool "lcd I30"
 
+config LCD_ANDROIDCOMPUTER
+        bool "rk3188 android computer lcd"
+
 config LCD_TL5001_MIPI
 	bool "TL5001 720X1280"
 
diff --git a/drivers/video/rockchip/screen/Makefile b/drivers/video/rockchip/screen/Makefile
index 40248ecdb6fc..c969ea1229ed 100644
--- a/drivers/video/rockchip/screen/Makefile
+++ b/drivers/video/rockchip/screen/Makefile
@@ -66,6 +66,7 @@ obj-$(CONFIG_LCD_RK3028_86V)   += lcd_E242868_rk3028_86v.o
 obj-$(CONFIG_LCD_MQ0801D)    += lcd_mq0801d.o
 obj-$(CONFIG_LCD_LD089WU1_MIPI)   += lcd_LD089WU1_mipi.o
 obj-$(CONFIG_LCD_B080XAN02_MIPI)   += lcd_B080XAN02_mipi.o
+obj-$(CONFIG_LCD_ANDROIDCOMPUTER)   += lcd_androidcomputer.o
 
 
 quiet_cmd_gen = GEN     $@
diff --git a/include/linux/input.h b/include/linux/input.h
index 97a7d17a4279..a207923f3612 100755
--- a/include/linux/input.h
+++ b/include/linux/input.h
@@ -440,7 +440,6 @@ struct input_keymap_entry {
 
 #define KEY_WIMAX		246
 #define KEY_RFKILL		247	/* Key that controls all radios */
-#define KEY_MODE_SWITCH		248	/* Key that controls all radios */
 
 /* Code 255 is reserved for special needs of AT keyboard driver */
 
diff --git a/include/linux/scaler-core.h b/include/linux/scaler-core.h
index cc9d36d42776..31278d0b14c5 100644
--- a/include/linux/scaler-core.h
+++ b/include/linux/scaler-core.h
@@ -156,7 +156,7 @@ int init_scaler_chip(struct scaler_chip_dev *chip, struct scaler_platform_data *
 int register_scaler_chip(struct scaler_chip_dev *chip);
 int unregister_scaler_chip(struct scaler_chip_dev *chip);
 
-#define SCALER_IOCTL_MAGIC 's'
+#define SCALER_IOCTL_MAGIC 'a'
 #define SCALER_IOCTL_POWER _IOW(SCALER_IOCTL_MAGIC, 0x00, char)
 #define SCALER_IOCTL_GET_CUR_INPUT _IOR(SCALER_IOCTL_MAGIC, 0x01, int)
 #define SCALER_IOCTL_SET_CUR_INPUT _IOW(SCALER_IOCTL_MAGIC, 0x02, int)
-- 
2.35.3

